application/web/scripts/homepage/homepage.js

```diff
@@ -113,75 +113,116 @@ function openEntryforEdit() {
  * This function is currently identical to cancelEntry() but will have more functionality added in the future.
  */
-function saveCurrentEntry() {
+async function saveCurrentEntry() {
     // Hide the text editor
     hideTextEditor();
 
     // Get the title text area and extract the title
     const titleTextArea = document.querySelector('#title-input');
-    const title = titleTextArea.value;
+    const entryTitle = titleTextArea.value;
     titleTextArea.style.display = '';
 
     // Get the entry text area and extract the entry content
-    const entryTextArea = document.querySelector('.entry-textarea');
-    const entry = entryTextArea.value;
+    // TODO: make selector cleaner
+    // const entryTextSelectorString = `#entry-textarea > div > div.CodeMirror-scroll > div.CodeMirror-sizer > div > div > div > div.CodeMirror-code > div > pre > span`;
+    const entryTextArea = document.querySelector('.CodeMirror').CodeMirror;
+    const entryContent = entryTextArea.getValue('\n');
     entryTextArea.value = '';
 
     // Get the list of past entries and create a new button for the current entry
     const buttonList = document.querySelector('.past-entries');
     const newEntryButton = document.createElement('button');
-    newEntryButton.innerText = title;
-
-    // Create an article element to display the entry content
-    const article = document.createElement('article');
-    article.innerText = entry;
-    article.style.display = 'none';
-
-    // Append the new entry button and article to the list of past entries
-    buttonList.append(newEntryButton);
-    buttonList.append(article);
-
-    // TODO: Replace with function to load entry from storage
-    newEntryButton.addEventListener('click', (event) => {
-        // article.style.display = 'block';
-        // titleTextArea.value = newEntryButton.innerText;
-        // const entryContent = document.querySelector('.CodeMirror-line');
-        // entryTextArea.value = entryContent.innerText;
-        // openEntryforEdit();
-        editJournal(event);
-    });
-
-    async function editJournal(event){
-        try {
-            const journals = await api.getJournals();
-            const title = event.target.innerText;
-    
-            for(const journal of journals){
-                const entries = await journal.getEntries();
-                const matchingEntry = entries.find(entry => entry.name === title);
-    
-                if(matchingEntry){
-                    const content = await matchingEntry.getContent();
-    
-                    // const titleTextArea = document.querySelector('#title-input');
-                    // titleTextArea.value = entryTitle
-    
-                    const entryTextArea = document.querySelector('.entry-textarea');
-                    entryTextArea.value = content;
-    
-                    break;
-                }
-            }
-        }catch(error) {
-            console.error(`An error occured: ${error}`);
-        }
-    }
```

Comment:
Is there a reason why you're removing this? Cuz I think your functionality and the editing functionality is different?

Reply:
The function didn't work and I think it should be done in a separate branch. Also having the function in another function seemed like bad practice.

Reply:
I think it was merged to main for this, but it seems like Ben found a better solution so I guess it's okay